### PR TITLE
add redis streams messaging to `redis-streams`

### PR DESCRIPTION
**Pull request by zzstoatzz**
stacked on https://github.com/PrefectHQ/prefect/pull/16416

supersedes https://github.com/PrefectHQ/prefect/pull/16415

### CODE REVIEW

File: src/prefect/server/utilities/messaging/__init__.py [hunk 1]
```diff
@@ -131,14 +131,21 @@ class CacheModule(Protocol):
     Cache: Type[Cache]
 
 
-def create_cache() -> Cache:
+def create_cache(topic: Optional[str] = None) -> Cache:
```
Comment (desertaxle) [inline src/prefect/server/utilities/messaging/__init__.py:134]: I think we should hold off on adding this kwarg for now. There isn't a convenient way for users to set this right now, and I think we can get away with a sane default until users request a custom topic for the cache.

File: src/integrations/prefect-redis/prefect_redis/messaging.py [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+M = TypeVar("M", bound=Message)
+
+MESSAGE_DEDUPLICATION_LOOKBACK = timedelta(minutes=5)
+
+
+class Cache(_Cache):
+    def __init__(self, topic: Optional[str] = None):
+        self.topic = topic or "default"
+        self._client: Redis = get_async_redis_client()
```
Comment (desertaxle) [inline src/integrations/prefect-redis/prefect_redis/messaging.py:41]: In conjunction with my comment about the `create_cache` kwarg, I think we need a more unique default topic name. Here's a suggestion, but there are many other valid options.
```suggestion
    def __init__(self, topic: str = "messaging-cache"):
        self.topic = topic
        self._client: Redis = get_async_redis_client()
```

**Comment by zzstoatzz** [comment]
makes sense @desertaxle! moved `topic` down to the redis specific init and changed `default` to `messaging-cache`

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**